File: server/src/utils/response.js
function sendSuccess(res, data = null, message = 'success', code = 0) {
  return res.json({
    code,
    data,
    message,
  });
}

function sendError(res, message = 'error', code = 500, status = 500, data = null) {
  return res.status(status).json({
    code,
    data,
    message,
  });
}

module.exports = {
  sendError,
  sendSuccess,
};

File: server/src/response.rs
use serde::Serialize;

#[derive(Serialize)]
pub struct ApiResponse<T>
where
    T: Serialize,
{
    pub code: u16,
    pub data: T,
    pub message: String,
}

impl<T> ApiResponse<T>
where
    T: Serialize,
{
    pub fn success(data: T, message: impl Into<String>) -> Self {
        Self {
            code: 0,
            data,
            message: message.into(),
        }
    }

    pub fn error(code: u16, data: T, message: impl Into<String>) -> Self {
        Self {
            code,
            data,
            message: message.into(),
        }
    }
}
